.changeset/large-spiders-protect.md

```diff
@@ -0,0 +1,15 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Hds::Table`
+- Added `@selectableColumnKey` argument which enables sorting by row selection state and specifies the corresponding selection state key.
+
+`Hds::Table::Tr`
+- Added `@selectableColumnKey` argument which enables sorting by row selection state and specifies the corresponding selection state key.
+- Added `@sortBySelectedOrder` (`undefined` | `asc` | `desc`) argument which determines the state of the sort button in the selected item column.
```

Comment:
[suggestion] we usually don't list the values of an enum in the release notes, unless necessary
```suggestion
- Added `@sortBySelectedOrder` argument which determines the state of the sort button in the selected item column.
```